### ISSUE TITLE
How long does it require to train on all 4 datasets?

### ISSUE THREAD

**Issue by ljquality** [STVIR/pysot#266]
I train the model with 4 GPU and find that the speed is very slow (about 1.8s/iter), so I want to know whether there is any problem? How long does it require to train on all 4 datasets?

**Comment by yc-zc** [comment]
What is your backbone?

**Comment by yc-zc** [comment]
Sorry, I mistakenly closed this issue

**Comment by ljquality** [comment]
> What is your backbone?

AlexNet. This is the config I adopt:
META_ARC: "siamrpn_alex_dwxcorr"

BACKBONE:
    TYPE: "alexnet"
    KWARGS:
        width_mult: 1.0   
    # PRETRAINED: 'pretrained_models/alexnet-bn.pth'
    TRAIN_LAYERS: ['layer4', 'layer5']
    TRAIN_EPOCH: 10
    LAYERS_LR: 1.0

ADJUST:
    ADJUST: False

RPN:
    TYPE: 'DepthwiseRPN'
    KWARGS:
        anchor_num: 5
        in_channels: 256
        out_channels: 256

MASK:
    MASK: False

ANCHOR:
    STRIDE: 8
    RATIOS: [0.33, 0.5, 1, 2, 3]
    SCALES: [8]
    ANCHOR_NUM: 5

TRACK:
    TYPE: 'SiamRPNTracker'
    PENALTY_K: 0.16
    WINDOW_INFLUENCE: 0.40
    LR: 0.30
    EXEMPLAR_SIZE: 127
    INSTANCE_SIZE: 287
    BASE_SIZE: 0
    CONTEXT_AMOUNT: 0.5

TRAIN:
    EPOCH: 50
    START_EPOCH: 0
    BATCH_SIZE: 32
    BASE_SIZE: 0
    OUTPUT_SIZE: 17
    BASE_LR: 0.00125
    CLS_WEIGHT: 1.
    LOC_WEIGHT: 1.2
    RESUME: ''

    LR:
        TYPE: 'log'
        KWARGS:
            start_lr: 0.01
            end_lr: 0.0005
    LR_WARMUP:
        TYPE: 'step'
        EPOCH: 5
        KWARGS:
            start_lr: 0.005
            end_lr: 0.01
            step: 1

DATASET:
    NAMES:
    - 'COCO'


    TEMPLATE:
        SHIFT: 4
        SCALE: 0.05
        BLUR: 0.0
        FLIP: 0.0
        COLOR: 1.0

    SEARCH:
        SHIFT: 64
        SCALE: 0.18
        BLUR: 0.2
        FLIP: 0.0
        COLOR: 1.0

    NEG: 0.05
    GRAY: 0.0

**Comment by yc-zc** [comment]
> > What is your backbone?
> 
> AlexNet. This is the config I adopt:
> META_ARC: "siamrpn_alex_dwxcorr"
> 
> BACKBONE:
> TYPE: "alexnet"
> KWARGS:
> width_mult: 1.0
> # PRETRAINED: 'pretrained_models/alexnet-bn.pth'
> TRAIN_LAYERS: ['layer4', 'layer5']
> TRAIN_EPOCH: 10
> LAYERS_LR: 1.0
> 
> ADJUST:
> ADJUST: False
> 
> RPN:
> TYPE: 'DepthwiseRPN'
> KWARGS:
> anchor_num: 5
> in_channels: 256
> out_channels: 256
> 
> MASK:
> MASK: False
> 
> ANCHOR:
> STRIDE: 8
> RATIOS: [0.33, 0.5, 1, 2, 3]
> SCALES: [8]
> ANCHOR_NUM: 5
> 
> TRACK:
> TYPE: 'SiamRPNTracker'
> PENALTY_K: 0.16
> WINDOW_INFLUENCE: 0.40
> LR: 0.30
> EXEMPLAR_SIZE: 127
> INSTANCE_SIZE: 287
> BASE_SIZE: 0
> CONTEXT_AMOUNT: 0.5
> 
> TRAIN:
> EPOCH: 50
> START_EPOCH: 0
> BATCH_SIZE: 32
> BASE_SIZE: 0
> OUTPUT_SIZE: 17
> BASE_LR: 0.00125
> CLS_WEIGHT: 1.
> LOC_WEIGHT: 1.2
> RESUME: ''
> 
> ```
> LR:
>     TYPE: 'log'
>     KWARGS:
>         start_lr: 0.01
>         end_lr: 0.0005
> LR_WARMUP:
>     TYPE: 'step'
>     EPOCH: 5
>     KWARGS:
>         start_lr: 0.005
>         end_lr: 0.01
>         step: 1
> ```
> 
> DATASET:
> NAMES:
> - 'COCO'
> 
> ```
> TEMPLATE:
>     SHIFT: 4
>     SCALE: 0.05
>     BLUR: 0.0
>     FLIP: 0.0
>     COLOR: 1.0
> 
> SEARCH:
>     SHIFT: 64
>     SCALE: 0.18
>     BLUR: 0.2
>     FLIP: 0.0
>     COLOR: 1.0
> 
> NEG: 0.05
> GRAY: 0.0
> ```

I'm not 100% sure but I beliebe something is wrong, I think what you expect is around 10 times faster. Could you provide a screenshot or low on how long each part took?

**Comment by ljquality** [comment]
> I'm not 100% sure but I believe something is wrong, I think what you expect is around 10 times faster. Could you provide a screenshot or logs on how long each part took?

![image](https://user-images.githubusercontent.com/25917120/71142699-1fd67600-2253-11ea-8ddf-f6c775abf254.png)
![image](https://user-images.githubusercontent.com/25917120/71142712-2c5ace80-2253-11ea-9576-91fb1dad80c7.png)
When the training step goes up to 3000, the speed is faster (no more than 1s/iter).  How long does it require with the default config on 16 GPU  in your experiments?

**Comment by ljquality** [comment]
> I'm not 100% sure but I believe something is wrong, I think what you expect is around 10 times faster. Could you provide a screenshot or logs on how long each part took?
![image](https://user-images.githubusercontent.com/25917120/71143173-9031c700-2254-11ea-80c8-214463b1d04f.png)
![image](https://user-images.githubusercontent.com/25917120/71143177-96c03e80-2254-11ea-8a5a-a7db47320055.png)

When the training step goes up to 3000, the speed is faster (no more than 1s/iter).  How long does it require with the default config on 16 GPU in your experiments?

**Comment by yc-zc** [comment]
My code has a lot of differences compare to what author provided, and since we used different strategy, I don't think my running time is meaningful. FYI it's around 1.3s/step, consider the different GPU we used I would argue it's reasonable. I will need to look into it to give you an answer. In the meantime, I would personally suggest you use a much bigger batch size. I mean, come on, you have 176GB GPU memory available but your batch size is only 32... I would say at least double it, and if it is me, I may choose something like 128 or even 256 lol

**Comment by ny222333** [comment]
Would you like to share the youtube dataset   preprocessed   with me？I find the link provided is expired.Thank you!

**Comment by zyc-ai** [comment]
Since no further question is asked, I'm closing it for now. 
Please open a new issue should you have other questions.

**Comment by wWHWw** [comment]
> > I'm not 100% sure but I believe something is wrong, I think what you expect is around 10 times faster. Could you provide a screenshot or logs on how long each part took?
> > ![image](https://user-images.githubusercontent.com/25917120/71143173-9031c700-2254-11ea-80c8-214463b1d04f.png)
> > ![image](https://user-images.githubusercontent.com/25917120/71143177-96c03e80-2254-11ea-8a5a-a7db47320055.png)
> 
> When the training step goes up to 3000, the speed is faster (no more than 1s/iter). How long does it require with the default config on 16 GPU in your experiments?

Sorry to reopen this issue.

I noticed your GPU-Util is around 100%, to my understanding that means your GPU is doing calculation in full capacity.
But in my own environment, my GPU-Util is around 0% , 100% pop up about every 5-10 seconds.
My speed is only 8-9s/iter and ETA is around 30 days.
Though I only have two GPU (2080TI and Titan xp) , but it shouldn't take this long.
Any chance you could share your hardware confituration?

Anthor question is will mechanical hard disk affect training speed drastically? Beacuse I find lots iowait during training from `iostat -x 2 20`@zyc-ai

**Comment by zyc-ai** [comment]
> > > I'm not 100% sure but I believe something is wrong, I think what you expect is around 10 times faster. Could you provide a screenshot or logs on how long each part took?
> > > ![image](https://user-images.githubusercontent.com/25917120/71143173-9031c700-2254-11ea-80c8-214463b1d04f.png)
> > > ![image](https://user-images.githubusercontent.com/25917120/71143177-96c03e80-2254-11ea-8a5a-a7db47320055.png)
> > 
> > 
> > When the training step goes up to 3000, the speed is faster (no more than 1s/iter). How long does it require with the default config on 16 GPU in your experiments?
> 
> I noticed your GPU-Util is around 100%, to my understanding that means your GPU is doing calculation in full capacity.
> But in my own environment, my GPU-Util is around 0% , 100% pop up about every 5-10 seconds.
> My speed is only 8-9s/iter and ETA is around 30 days.
> Though I only have two GPU (2080TI and Titan xp) , but it shouldn't take this long.
> Any chance you could share your hardware confituration?
> 
> Anthor question is will mechanical hard disk affect training speed drastically? Beacuse I find lots iowait during training from `iostat -x 2 20`@zyc-ai

You really should have starts from discussions, I mean, closed issues are usually forgotten. 
My hardware configuration is 8*1080ti. And I believe in your case, the training spped is more limited by your hard drive or CPU. 0% of GPU-Util suggests that your GPU is waiting for data to perform forward. So, use SSD, and if it's impractical, more workers for dataloader
